pkg/v21/chartconfig/current.go

```diff
@@ -36,15 +38,22 @@ func (c *ChartConfig) GetCurrentState(ctx context.Context, clusterConfig Cluster
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
+
 	chartConfigList, err := cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs(resourceNamespace).List(listOptions)
 	if tenant.IsAPINotAvailable(err) {
 		c.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available yet")
 		c.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 		resourcecanceledcontext.SetCanceled(ctx)
 		return nil, nil
-
 	} else if err != nil {
 		return nil, microerror.Mask(err)
+	} else if errors.Is(ctx.Err(), context.DeadlineExceeded) {
```

Comment:
I don't think `else` is needed here. And possible confusing. Not a blocker though.

Reply:
Thx! Good call, done.